src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -181,4 +179,34 @@ void CASEServer::OnSessionEstablished(const SessionHandle & session)
                     ChipLogValueScopedNodeId(session->GetPeer()));
     PrepareForSessionEstablishment(session->GetPeer());
 }
+
+void CASEServer::SendBusyStatusReport(Messaging::ExchangeContext * ec)
+{
+    using namespace Protocols::SecureChannel;
+
+    // 500 milliseconds is an example value mentioned in specification section "4.10.1.5. Busy"
+    uint16_t minWaitTime    = 500; // milliseconds
+    auto protocolDataHandle = System::PacketBufferHandle::NewWithData(&minWaitTime, sizeof(minWaitTime));
+    VerifyOrReturn(!protocolDataHandle.IsNull(), ChipLogError(SecureChannel, "Failed to allocate protocol data for status report"));
+
+    StatusReport statusReport(GeneralStatusCode::kBusy, Id, kProtocolCodeBusy, std::move(protocolDataHandle));
+
+    auto handle = System::PacketBufferHandle::New(statusReport.Size());
+    VerifyOrReturn(!handle.IsNull(), ChipLogError(SecureChannel, "Failed to allocate status report message"));
+    Encoding::LittleEndian::PacketBufferWriter bbuf(std::move(handle));
+
+    statusReport.WriteToBuffer(bbuf);
+
+    System::PacketBufferHandle msg = bbuf.Finalize();
+    VerifyOrReturn(!msg.IsNull(), ChipLogError(SecureChannel, "Failed to allocate status report message"));
```

Comment:
Isn't this a "failed to fit", not "failed to finalize"?

Reply:
Sorry,  cut-n-paste, error, fixed!